cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +156,25 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if opts.VerityHashDevice != "" && opts.VerityRootHash == "" {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a root hash was not specified", what, where)
+	}
+	if opts.VerityRootHash != "" && opts.VerityHashDevice == "" {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a hash device was not specified", what, where)
+	}
+	if opts.VerityHashOffset != 0 && (opts.VerityHashDevice == "" || opts.VerityRootHash == "") {
+		return fmt.Errorf("cannot mount %q at %q: mount with dm-verity was requested but a hash device and root hash were not specified", what, where)
+	}
+	if opts.VerityHashDevice != "" && opts.VerityRootHash != "" {
+		options = append(options, fmt.Sprintf("verity.roothash=%s", opts.VerityRootHash))
+
+		escaper := strings.NewReplacer(",", `\,`, " ", `\ `)
+		options = append(options, fmt.Sprintf("verity.hashdevice=%s", escaper.Replace(opts.VerityHashDevice)))
```

Comment:
Would that work to use `"verity.hashdevice=\"%s\""`

Reply:
Instead of escaping.

Reply:
- I reverted the escaping logic.
- I escaped the hash device using double quotes.
- I forbid backslashes, commas, spaces, colons, double quotes and spaces from appearing in paths used for dm-verity hash devices.
- added a test.